### PR TITLE
Revise v1.6 migration guide

### LINKED ISSUE #3696: New flag `--defer-state` for more-granular control of state used for deferral versus logic comparison
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/node-selection/syntax#about-node-selection
https://docs.getdbt.com/reference/node-selection/defer

### What part(s) of the page would you like to see updated?

Thorough context in the description of this issue:
- https://github.com/dbt-labs/dbt-core/issues/7300

Big idea is, you can now pass two manifests if you want to:
- one for use in node selection, i.e. `state:modified` — this is `--state`, and roughly maps to "logical" state
- one for use by `--defer` (and now `clone`) — this is `--defer-state` if specified, and roughly maps to "applied" state

In >90% of cases, we expect to resolve both using the same state/manifest.

(For context on what I mean by "logical" versus "applied" state: [docs](https://docs.getdbt.com/docs/dbt-cloud-apis/project-state#definition-logical-vs-applied-state-of-dbt-nodes); [internal notion](https://www.notion.so/dbtlabs/Logical-definition-vs-applied-project-state-d3892048faac4afdb36b4369bf198eda))

### Additional information

- https://github.com/dbt-labs/dbt-core/pull/7690

### LINKED ISSUE #3723: `dbt docs generate --empty-catalog` to skip catalog generation
### Contributions

- [x] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/commands/cmd-docs#dbt-docs-generate

### What part(s) of the page would you like to see updated?

Add `--empty-catalog` flag

### Additional information

https://github.com/dbt-labs/dbt-core/issues/6980
https://github.com/dbt-labs/dbt-core/pull/7202

### PR DESCRIPTION
**Pull request by jtcohen6**
resolves #3696
resolves #3723

Revise v1.6 migration guide for readability. Still a handful of TODOs.

### Materialized views
- Add `VersionBlock` components for adapter MVs, missed in https://github.com/dbt-labs/docs.getdbt.com/pull/3593
- We seem to be missing docs on Snowflake dynamic tables. I don't know why https://github.com/dbt-labs/docs.getdbt.com/issues/3494 was closed.

### Other updates
- Add manifest v10 (will need addition to schemas.getdbt.com: https://github.com/dbt-labs/schemas.getdbt.com/issues/24)
- Add note to upgrade guide for `state:old` and `state:unmodified` (missed in https://github.com/dbt-labs/docs.getdbt.com/pull/3596). I haven't done a thorough comb-through of other closed v1.6 PRs to see what needs addition to the migration guide.

### CODE REVIEW

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md
Comment (mirnawong1) [inline website/docs/guides/migration/versions/01-upgrading-to-v1.6.md:41]: new and revamped SL docs

```suggestion
- [Revamped dbt Semantic Layer pages](/docs/use-dbt-semantic-layer/dbt-sl)
- [dbt Semantic Layer API](/docs/use-dbt-semantic-layer/sl-api-overview)
- [dbt Semantic Layer manifest](/docs/use-dbt-semantic-layer/sl-manifest)
- [Legacy dbt Semantic Layer migration guide](guides/migration/sl-migration)
```

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md
Comment (jtcohen6) [inline website/docs/guides/migration/versions/01-upgrading-to-v1.6.md:30]: Discussion for adapter maintainers: https://github.com/dbt-labs/dbt-core/discussions/7958

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md
Comment (jtcohen6) [inline website/docs/guides/migration/versions/01-upgrading-to-v1.6.md]: After #3742 is merged:
```suggestion
[`dbt clone`](/reference/commands/clone) leverages each data platform's functionality for creating lightweight copies of dbt models from one environment into another. Useful when quickly spinning up a new development environment, or promoting specific models from a staging environment into production.
```

File: website/dbt-versions.js
Comment (jtcohen6) [inline website/dbt-versions.js]:
```suggestion
    EOLDate: "2024-07-31",
```

File: website/docs/guides/migration/versions/01-upgrading-to-v1.6.md
Comment (jtcohen6) [inline website/docs/guides/migration/versions/01-upgrading-to-v1.6.md]:
```suggestion
dbt Core v1.6 is available as a release candidate. Final release planned for July 31.
```

**Review by graciegoheen — APPROVED** [review]

**Pull request merged**
